### DISCUSSION TITLE
关于连续查询的一些问题

### DISCUSSION THREAD

**Discussion by StrongManLi** [apache/iotdb#10653] [Q&A] [answered]
1.我下面的 SQL 执行之后，在另存的表中，时间戳都是 1970-01-01T0:00:00这是由什么导致的呢，我应该如何设置时间呢
```sql 
create CQ hourly resample every 1h boundary 2023-07-07T00:00:00 begin select sum(temperature) into root.hourly.::
(temperature) from root.a.b.* align by device end
```
2.聚合函数的查询结果起一个别名，能在where条件里对这个别名进行筛选吗
3.连续查询能每5分钟统计一小时内的总和吗？
看了半天文档了，但是由于太菜，还是有问题没解决，特此请教大佬们，谢谢。

**Comment by jixuan1989** [comment]
@stefaniexin @JackieTien97   have a look about this feature?

**Comment by JackieTien97** [comment, marked as answer]
1. 第一个问题你可以试着加上group by clause
2. 第二个问题，where里是不能对聚合结果进行过滤的，必须用having子句进行过滤，比如说 `select sum(s1) as sum_alias from root.db.d1 having sum_alias > 1`， 你可以参考官网https://iotdb.apache.org/zh/UserGuide/V1.1.x/Query-Data/Having-Condition.html;
3. 可以的，`create CQ hourly resample every 5m RANGE 1h boundary 2023-07-07T00:00:00 begin select sum(temperature) into root.hourly.::
(temperature) from root.a.b.* group by(1h) align by device end`你也可以参考官网文档 https://iotdb.apache.org/zh/UserGuide/V1.1.x/Query-Data/Continuous-Query.html#%E8%BF%9E%E7%BB%AD%E6%9F%A5%E8%AF%A2%E7%9A%84%E7%94%A8%E4%BE%8B

**Reply by StrongManLi** [reply]
谢谢
